src/verify.c

```diff
@@ -770,6 +770,51 @@ static enum swupd_code deal_with_extra_files(struct manifest *manifest, bool fix
 	return ret;
 }
 
+static struct list *sanitize_files(struct list *bundles_files, struct list *all_files)
+{
+	struct list *list_all = list_head(all_files);
+	struct list *list_bundles = list_head(bundles_files);
+	struct list *sanitized = NULL;
+	struct file *filea, *fileb;
+
+	/* the list of all_files will always include all files in bundles_files
+	 * and both lists are sorted by filename */
+	while (list_all) {
+
+		if (!list_bundles) {
+			/* we have reached the last element of the list of bundle files
+			 * no need to keep going */
+			return list_head(sanitized);
```

Comment:
It's better just to save the first element of the list to return so we don't need to run through the list again to find the first element.

Reply:
good idea